### ISSUE TITLE
Allow URL params in `$url()` RPC

### ISSUE THREAD

**Issue by douglasduteil** [honojs/hono#1870]
### What is the feature you are proposing?

I love the [`$url()`](https://hono.dev/guides/rpc#url) feature.
It just that I don't find the returned `mergePath` very useful.
Can it use the `replaceUrlParam` too if the url has input params ?

I'm thinking about some like this :

```ts
const router = new Hono()
  .get(
    "/foo/:id/bar",
    zValidator("param", z.object({ id: z.string() })),
    function ({ text, req }) {
	  const { id } = req.valid("params");
	  // [...]
      return text("OK");
    },
  );

// [...]

app_hc.foo[":id"].bar.$url({ param: { id: "1" } });
// >> http://localhost:3000/foo/1/bar
```

**Comment by yusukebe** [comment]
Hi @douglasduteil 

Sorry for the delayed response, and thank you for the suggestion. This is cool!

I've implemented this feature in PR #1885. If you can, could review or see it?

**Comment by douglasduteil** [comment]
Closed with #1885
